src/Console/Shell.php

```diff
@@ -472,12 +472,64 @@ public function runCommand($argv, $autoMethod = false, $extra = [])
             return $this->main(...$this->args);
         }
 
-        $this->out($this->OptionParser->help($command));
+        $this->outputUnknownCommand($command);
 
         return false;
     }
 
     /**
+     * Output a message in the console stating that the command can not be found and tries to guess what the user
+     * wanted to say. Output a list of available subcommands as well.
+     *
+     * @param string $command Unknown command name trying to be dispatched.
+     * @return void
+     */
+    protected function outputUnknownCommand($command)
```

Comment:
I checked and it does not seem to.
The previous behavior (outputting the full help of the command) that this PR replace did this : 

```
$this->out($this->OptionParser->help($command));
```
https://github.com/cakephp/cakephp/pull/10941/files#diff-d606d81f2aa3b802daa770d43e96be2fL475